### ISSUE TITLE
a littile 

### ISSUE THREAD

**Issue by hlthu** [snsun/pit-speech-separation#14]
You said:
```
def get_padded_batch(file_list, batch_size, input_size, output_size,
                     num_enqueuing_threads=4, num_epochs=1, shuffle=True):
    """Reads batches of SequenceExamples from TFRecords and pads them.
    Can deal with variable length SequenceExamples by padding each batch to the
    length of the longest sequence with zeros.
    Args:
        file_list: A list of paths to TFRecord files containing SequenceExamples.
        batch_size: The number of SequenceExamples to include in each batch.
        input_size: The size of each input vector. The returned batch of inputs
            will have a shape [batch_size, num_steps, input_size].
        num_enqueuing_threads: The number of threads to use for enqueuing
            SequenceExamples.
    Returns:
        inputs: A tensor of shape [batch_size, num_steps, input_size] of floats32s.
        labels: A tensor of shape [batch_size, num_steps] of float32s.
        lengths: A tensor of shape [batch_size] of int32s. The lengths of each
            SequenceExample before padding.
    """
```
What's the meaning of the `lengths`. it said that it has the shape of [batch_size], do you mean that it is all the same for all batchs? Or is it a vector with the dim the same as the length of audio files?

I have searched the `decode()`, and it tell me that it is the shape of [batch_size].
So every first samples in every batch have the same length?

**Comment by snsun** [comment]
Hi，
Length is a vector contains the length of every utterance in this  minibatch. When LSTM is used, for every mini-batch, the utterances will be padded to the maximum length. Elements in this vector record the real length for every utterance before padding. This will be used when you construct a LSTM model.

Best
Sining

Sent from Mail for Windows 10

From: Lu Huang
Sent: 2018年5月29日 21:03
To: snsun/pit-speech-separation
Cc: Subscribed
Subject: [snsun/pit-speech-separation] a littile (#14)

You said:
def get_padded_batch(file_list, batch_size, input_size, output_size,
                     num_enqueuing_threads=4, num_epochs=1, shuffle=True):
    """Reads batches of SequenceExamples from TFRecords and pads them.
    Can deal with variable length SequenceExamples by padding each batch to the
    length of the longest sequence with zeros.
    Args:
        file_list: A list of paths to TFRecord files containing SequenceExamples.
        batch_size: The number of SequenceExamples to include in each batch.
        input_size: The size of each input vector. The returned batch of inputs
            will have a shape [batch_size, num_steps, input_size].
        num_enqueuing_threads: The number of threads to use for enqueuing
            SequenceExamples.
    Returns:
        inputs: A tensor of shape [batch_size, num_steps, input_size] of floats32s.
        labels: A tensor of shape [batch_size, num_steps] of float32s.
        lengths: A tensor of shape [batch_size] of int32s. The lengths of each
            SequenceExample before padding.
    """
What's the meaning of the lengths. it said that it has the shape of [batch_size], do you mean that it is all the same for all batchs? Or is it a vector with the dim the same as the length of audio files?
I have searched the decode(), and it tell me that it is the shape of [batch_size].
So every first samples in every batch have the same length?
